### PR TITLE
feat: consumer URL

### PR DESCRIPTION
**Pull request by miton18**

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+    getConsumerWebsocketPulsarUrlWithTopic 
+} = require('./config');
 
-module.exports = (topic) => new WebSocket(getWebsocketPulsarUrlWithTopic(topic), {
-    headers: {
-        Authorization: `Bearer ${process.env.ADDON_PULSAR_TOKEN}`,
-    },
-});
\ No newline at end of file
+module.exports = (topic, subscription=null) => {
```
Comment (gaelreyrol) [inline src/client.js]:
```suggestion
module.exports = (topic, subscription = null) => {
```

File: src/client.js [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
+    let url;
```
Comment (gaelreyrol) [inline src/client.js]: url can default to `getProducerWebsocketPulsarUrlWithTopic(topic)`

File: src/client.js [hunk 3]
```diff
@@ -1,9 +1,23 @@
...
+    if (subscription) {
+        url = getConsumerWebsocketPulsarUrlWithTopic(topic, subscription)
+    } else {
```
Comment (gaelreyrol) [inline src/client.js]: Remove `else` block

File: src/config.js [hunk 4]
```diff
@@ -1,14 +1,18 @@
 require('dotenv').config();
 
-function getWebsocketPulsarUrl() {
+function getWebsocketPulsarUrl(role='consumer') {
```
Comment (gaelreyrol) [inline src/config.js]:
```suggestion
function getWebsocketPulsarUrl(role = 'consumer') {
```

File: src/config.js [hunk 5]
```diff
@@ -1,14 +1,18 @@
...
+    return `wss://${hostname}:${port}/ws/v2/${role}/persistent/${tenant}/${namespace}`;
 }
 
-module.exports.getWebsocketPulsarUrlWithTopic = function (topic) {
-    return getWebsocketPulsarUrl() + `/${topic}`;
+module.exports.getProducerWebsocketPulsarUrlWithTopic = function (topic) {
+    return getWebsocketPulsarUrl('producer') + `/${topic}`;
+}
+
+module.exports.getConsumerWebsocketPulsarUrlWithTopic = function (topic, subscription = '') {
```
Comment (gaelreyrol) [inline src/config.js]: `susbscription` can be empty ?

**Review by gaelreyrol — CHANGES_REQUESTED** [review]

File: src/config.js (hunk 5 above)
Comment (miton18) [reply src/config.js]: hum... no I don't think so
https://pulsar.apache.org/docs/en/client-libraries-websocket/#consumer-endpoint

**Review by gaelreyrol — APPROVED** [review]

**Pull request merged**